src/Dotnet.Script.DependencyModel/ProjectSystem/ScriptProjectProvider.cs

```diff
@@ -60,12 +61,24 @@ public string CreateProject(string targetDirectory, string defaultTargetFramewor
                 return null;
             }
 
-            _logger.Debug($"Creating project file for *.csx files found in {targetDirectory} using {defaultTargetFramework} as the default framework." );
-            
+            _logger.Debug($"Creating project file for *.csx files found in {targetDirectory} using {defaultTargetFramework} as the default framework.");
+
             var csxFiles = Directory.GetFiles(targetDirectory, "*.csx", SearchOption.AllDirectories);
+            return CreateProjectFileFromScriptFiles(targetDirectory, defaultTargetFramework, csxFiles);
+        }
+
+        public string CreateProjectForScriptFile(string scriptFile)
```

Comment:
can we delete the old one now? `public string CreateProject(string targetDirectory, string defaultTargetFramework = "net46", bool enableNuGetScriptReferences = false)`?

Reply:
No, that one is still used from OmniSharp where still need to look at the folder structure as a whole.
You can see that it is used from 2 different places in dotnet-script. For instance in interactive mode where there is no script file.

Reply:
https://github.com/filipw/dotnet-script/blob/033b71b22ab51768b33697d1fa4598b6c7ef0643/src/Dotnet.Script.Core/ScriptCompiler.cs#L117

Reply:
So what I would like to get rid of is this `Directory.GetFiles(targetDirectory, "*.csx", SearchOption.AllDirectories)` search.
I thin we can still do it.

 - The usage in interactive mode can be treated differently. We don't need to scan for any CSX there, we really just want to create a temporary csproj so that the base runtime assemblies for .NET Core are loaded (and then in case new nuget references are added)
 - For OmniSharp, OmniSharp itself will discover all CSX files for us, so it could just pass a list of those, and we could set projects accordingly. 

Or maybe I am missing something?
This is somewhat unrelated to this PR, but fundamentally I'd like to not have this scan all behavior.

Reply:
I agree. Maybe we could do that in a separate PR? This fixes the 99% scenario where we actually execute a script located on disc. Every other scenario is left unchanged. And then we try to align the other overload with OmniSharp in a separate PR